src/main/java/haveric/recipeManager/RecipeManager.java

```diff
@@ -175,8 +176,15 @@ public void reload(CommandSender sender, boolean check, boolean firstTime) {
             }
 
             if (!firstTime && !Settings.getInstance().getClearRecipes()) {
-                Vanilla.restoreAllButSpecialRecipes();
-                Recipes.getInstance().index.putAll(Vanilla.initialRecipes);
+                if (!Version.has1_12Support()) {
+                    Vanilla.restoreAllButSpecialRecipes();
+                    Recipes.getInstance().index.putAll(Vanilla.initialRecipes);
+                } else {
+                    try {
+                        Bukkit.getServer().resetRecipes();
+                        Vanilla.init();
+                    } catch (NullPointerException npe) { /* Test Framework hates this */ }

```

Comment:
Odd, I'm not noticing any issues with the other one after removing the try catch.